remix-auth/app/routes/signin.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+  useLoaderData,
+  Link,
+  useSearchParams,
+  useActionData,
+} from "@remix-run/react";
+import { type ActionFunctionArgs, json, redirect } from "@remix-run/node";
+import { auth } from "~/services/auth.server";
+import { BackIcon, OAuthIcons } from "../icons";
+import SigninForm from "../components/auth/SigninForm";
+import { transformSearchParams } from "~/utils";
+
+export const loader = async () => {
+  const providers = await auth.getProvidersInfo();
+
+  return json({
+    providers,
+    OAuthUrls: providers.oauth.map((provider) =>
+      auth.getOAuthUrl(provider.name)
+    ),
```

Comment:
Same as the index page, we can call `getOAuthUrl` directly in the component with the client auth api.